### PR TITLE
[AC-1650] [AC-1578]

### PR DESCRIPTION
**Pull request by cturnbull-bitwarden**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by Thomas-Avery — DISMISSED** [review]
SM changes look good.

File: src/Core/Enums/PlanType.cs
Comment (Thomas-Avery) [inline src/Core/Enums/PlanType.cs:30]: Something to double-check, the display names for the `2020` plans are currently `2023`.

In the static store plans, the names use `2020`. Disregard if this intended.

**Comment by eliykat** [comment]
This will cause conflicts with #3164; also the improvements in that PR would avoid all the additional duplicated Plan objects here. I suggest either waiting until that is merged, or if you can't wait for that, can you please take over the subsequent reviews for #3164 as it will require additional changes if this is merged first.

**Comment by cturnbull-bitwarden** [comment]
> This will cause conflicts with #3164; also the improvements in that PR would avoid all the additional duplicated Plan objects here. I suggest either waiting until that is merged, or if you can't wait for that, can you please take over the subsequent reviews for #3164 as it will require additional changes if this is merged first.

Thanks for the heads-up, @eliykat! We had planned to sneak this in before @cyprain-okeke's refactor, but now we need to make additional refactors that would further block his work. We're going to switch up the priorities to get his work in first, then rebase this on top of his work. Moving this to be a draft for now

**Review by cyprain-okeke — COMMENTED** [review]
Looks Good

File: src/Core/Models/StaticStore/Plan.cs
Comment (cturnbull-bitwarden) [reply src/Core/Models/StaticStore/Plan.cs:31]: Yes, normally we would want to only set `LegacyYear` at instantiation. In this case, we need to update the value later depending on if the feature flag is enabled or not. If it's disabled, we need to display the old plans to the user instead of the new plans, and since `LegacyYear` determines which plans are displayed, it has to be null in that case.

**Review by cyprain-okeke — DISMISSED** [review]
LGTM

**Review by cyprain-okeke — APPROVED** [review]
Looks Good

**Review by Thomas-Avery — APPROVED** [review]
LGTM

**Review by cyprain-okeke — APPROVED** [review]

**Review by Thomas-Avery — DISMISSED** [review]

**Review by r-tome — APPROVED** [review]
Great work! Nice usage of transactions on the migration scripts.

**Review by cyprain-okeke — APPROVED** [review]

**Review by Thomas-Avery — APPROVED** [review]

**Review by cyprain-okeke — DISMISSED** [review]

**Review by r-tome — DISMISSED** [review]

**Review by Thomas-Avery — APPROVED** [review]

**Review by cyprain-okeke — APPROVED** [review]

**Review by r-tome — APPROVED** [review]

**Pull request merged**
